hack/generator/pkg/codegen/pipeline/convert_allof_and_oneof_to_objects_test.go

```diff
@@ -222,6 +224,40 @@ func TestMergeResourceMissingStatus(t *testing.T) {
 	g.Expect(synth.intersectTypes(r2, r1)).To(Equal(expected))
 }
 
+func TestMergeValidated(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	maxLen := int64(32)
+	validatedString := astmodel.NewValidatedType(
+		astmodel.StringType,
+		astmodel.StringValidations{
+			MaxLength: &maxLen,
```

Comment:
Why does this require a pointer to `maxLen`?

Reply:
it’s optional